packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -3253,11 +3255,17 @@ function beginWork(
       }
       return bailoutOnAlreadyFinishedWork(current, workInProgress, renderLanes);
     } else {
-      // An update was scheduled on this fiber, but there are no new props
-      // nor legacy context. Set this to false. If an update queue or context
-      // consumer produces a changed value, it will set this to true. Otherwise,
-      // the component will assume the children have not changed and bail out.
-      didReceiveUpdate = false;
+      if ((current.effectTag & ForceUpdateForLegacySuspense) !== NoEffect) {
```

Comment:
I wouldn't expect this check to be necessary, since whenever you hit this block, you'll always fall into the `updateSimpleMemoComponent` path, so you can do everything in there.

Reply:
Related: https://github.com/facebook/react/pull/19216/files#r447959102

Reply:
This isn't specific to SimpleMemo though. This block is to fix the issue for all other cases (e.g. normal memo, or a function component).

Reply:
Ah. God Legacy Mode is annoying. OK that makes sense then.